### PR TITLE
Implement comparison page from cna genes table, and add "Compare" button to tables

### PR DESCRIPTION
**Pull request by adamabeshouse**
Fixes a point in https://github.com/cBioPortal/cbioportal/issues/7831#issuecomment-706392418

image

I think it's better to wait for structural variant migration to be merged before working on the fusion table

### CODE REVIEW

File: src/pages/studyView/chartHeader/ChartHeader.tsx [hunk 1]
```diff
@@ -189,16 +190,19 @@ export class ChartHeader extends React.Component<IChartHeaderProps, {}> {
...
                     <a
                         className={classnames('dropdown-item', {
                             [styles.disabledMenuItem]:
                                 this.props.selectedRowsKeys!.length < 2,
                         })}
                         onClick={() => {
-                            const hugoGeneSymbols = this.props
-                                .selectedRowsKeys!;
-                            if (hugoGeneSymbols.length >= 2) {
+                            const hugoGeneSymbols = getHugoGeneSymbols(
```
Comment (kalletlak) [inline src/pages/studyView/chartHeader/ChartHeader.tsx:201]: move this into `if` block at line 205?

**Review by kalletlak — APPROVED** [review]

File: src/pages/studyView/chartHeader/ChartHeader.tsx (hunk 1 above)
Comment (adamabeshouse) [reply src/pages/studyView/chartHeader/ChartHeader.tsx:201]: good call

File: src/pages/studyView/StudyViewPageStore.ts [hunk 2]
```diff
@@ -874,6 +879,81 @@ export class StudyViewPageStore {
         });
     }
 
+    private createCnaGeneComparisonSession(
```
Comment (alisman) [inline src/pages/studyView/StudyViewPageStore.ts:882]: shall we this giant thing up into the comparison utils file above?

File: src/pages/studyView/StudyViewPageStore.ts [hunk 3]
```diff
@@ -874,6 +879,81 @@ export class StudyViewPageStore {
...
+                    );
+
+                    const genes = await client.fetchGenesUsingPOST({
+                        geneIdType: 'HUGO_GENE_SYMBOL',
+                        geneIds: hugoGeneSymbols,
+                    });
+
+                    const cnaByGeneAndAlteration = _.groupBy(
+                        (
+                            await client.fetchMolecularDataInMultipleMolecularProfilesUsingPOST(
```
Comment (alisman) [inline src/pages/studyView/StudyViewPageStore.ts:925]: kinda don't like the practice of embedding awaits inside of logic like this.  it just obscures the async nature.   i know asyn/await is supposed to smooth async coding patterns but i still think it's nice to assign it to a variable and then deal with that variable.   more explicit.

File: src/pages/studyView/StudyViewPageStore.ts [hunk 4]
```diff
@@ -874,6 +879,81 @@ export class StudyViewPageStore {
...
+                                        REQUEST_ARG_ENUM.PROJECTION_DETAILED,
+                                    molecularDataMultipleStudyFilter: {
+                                        entrezGeneIds: genes.map(
+                                            g => g.entrezGeneId
+                                        ),
+                                        sampleMolecularIdentifiers,
+                                    } as MolecularDataMultipleStudyFilter,
+                                }
+                            )
+                        ).filter(d => d.value && Math.abs(d.value) === 2), // filter out non-alterations
```
Comment (alisman) [inline src/pages/studyView/StudyViewPageStore.ts:937]: magic number.  what is 2?   maybe make a nice named constant.

**Comment by adamabeshouse** [comment]
@cBioPortal/product could you please take a look at this?

**Pull request merged**